src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/IllegalTypeCheck.java

```diff
@@ -236,6 +236,7 @@ private boolean isContainVerifiableType(DetailAST modifiers) {
                      modifier = modifier.getNextSibling()) {
```

Comment:
please do not do all optimizations in one PR. Split this request to separate issue